phoenicis-tools/src/main/java/org/phoenicis/tools/http/PhoenicisUrlConnection.java

```diff
@@ -0,0 +1,102 @@
+package org.phoenicis.tools.http;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.util.Map;
+
+/**
+ * Wrapper around {@link HttpURLConnection class} that follows correctly HTTP redirects
+ */
+class PhoenicisUrlConnection {
+    private Integer responseCode;
+    private HttpURLConnection delegateUrlConnexion;
+    private URL url;
+    private Map<String, String> headers;
+
+    private PhoenicisUrlConnection(HttpURLConnection urlConnection, URL url) {
```

Comment:
Add some JavaDoc to the methods (I know it's not required for private but it's also good for us to understand it).